### PR TITLE
Fix: Point2D serialization and deserialization

### PR DESCRIPTION
**Pull request by Emrehzl94**
Fixes #<Replace with the number of the issue, Mandatory>

One sentence summary of the change.

## Proposed Changes (Mandatory)

A brief list of proposed changes in order to fix the issue:

  -
  -
  -

### CODE REVIEW

File: common/src/test/kotlin/streams/utils/JSONUtilsTest.kt [hunk 1]
```diff
@@ -111,6 +111,81 @@ class JSONUtilsTest {
...
+        val cdcMap = mapOf<String, Any>(
+            "meta" to mapOf("timestamp" to timestamp,
+                "username" to "user",
+                "txId" to 1,
+                "txEventId" to 0,
+                "txEventsCount" to 1,
+                "operation" to OperationType.created),
+            "payload" to mapOf("id" to "0",
+                "before" to null,
+                "after" to NodeChange(properties = mapOf("location" to mapOf("crs" to "wgs-84", "latitude" to 12.78, "longitude" to 56.7, "height" to null)),
```
Comment (fbiville) [inline common/src/test/kotlin/streams/utils/JSONUtilsTest.kt:130]: is the "height" attribute required here? can we skip it?

File: common/src/test/kotlin/streams/utils/JSONUtilsTest.kt [hunk 2]
```diff
@@ -111,6 +111,81 @@ class JSONUtilsTest {
...
+                "operation" to OperationType.created),
+            "payload" to mapOf("id" to "0",
+                "before" to null,
+                "after" to NodeChange(properties = mapOf("location" to mapOf("crs" to "cartesian", "x" to 12.78, "y" to 56.7, "z" to null)),
+                    labels = listOf("LabelCDC")),
+                "type" to EntityType.node),
+            "schema" to mapOf("properties" to mapOf("location" to "PointValue"))
+        )
+
+        val cdcString = """{
```
Comment (fbiville) [inline common/src/test/kotlin/streams/utils/JSONUtilsTest.kt:173]: this (and its related assertions) can be moved to a separate test imo

File: common/src/main/kotlin/streams/utils/JSONUtils.kt [hunk 3]
```diff
@@ -236,7 +236,7 @@ abstract class StreamsTransactionEventDeserializer<EVENT, PAYLOAD: Payload> : Js
         ?.properties
         ?.mapValues {
             if (points.contains(it.key)) {
-                org.neo4j.values.storable.PointValue.fromMap((it.value as Map<String, Any>).toMapValue())
+                org.neo4j.values.storable.PointValue.fromMap((it.value as Map<String, Any>).toMapValue().filter { _, u -> u != Values.NO_VALUE })
```
Comment (fbiville) [inline common/src/main/kotlin/streams/utils/JSONUtils.kt:239]: you could instead update `streams.utils.JSONUtilsKt#toMapValue` to something like this:

```kotlin
fun Map<String, Any?>.toMapValue(): MapValue {
    return this
        .filterValues { v -> v != null }
        .entries
        .fold(MapValueBuilder()) { builder, entry -> 
            builder.add(entry.key, Values.of(entry.value, false)); builder}
        .build()
}
```

I even wonder if we need an extension function like this at all... `toMapValue` is only used in 2 tests and here. A regular function (or no function at all) would maybe work as well. 
Please lmk what you think :)

File: common/src/test/kotlin/streams/utils/JSONUtilsTest.kt (hunk 1 above)
Comment (Emrehzl94) [reply common/src/test/kotlin/streams/utils/JSONUtilsTest.kt:130]: It's actually part of the test case to see if we're able to deserialize it without having any issues. The reason I did it this way is even though we prevented serializer from generating null values, there might still be some old events generating height = null value, so we decided to cover this case as well.

File: common/src/test/kotlin/streams/utils/JSONUtilsTest.kt (hunk 2 above)
Comment (Emrehzl94) [reply common/src/test/kotlin/streams/utils/JSONUtilsTest.kt:173]: yes, this was also another idea, I just followed the pattern of the other unit tests in this class but of course separating them much cleaner I think.

File: common/src/main/kotlin/streams/utils/JSONUtils.kt (hunk 3 above)
Comment (Emrehzl94) [reply common/src/main/kotlin/streams/utils/JSONUtils.kt:239]: yes, this also crossed my mind but I think this is a general method that can also be used in other places to map values. This is why I decided to add the NO_VALUE filter in the point specific method. But of course getting rid of toMapValue method could also work.

File: common/src/test/kotlin/streams/utils/JSONUtilsTest.kt (hunk 2 above)
Comment (fbiville) [reply common/src/test/kotlin/streams/utils/JSONUtilsTest.kt:173]: let's split this test into 2 then, if that's ok with you?

File: common/src/main/kotlin/streams/utils/JSONUtils.kt (hunk 3 above)
Comment (fbiville) [reply common/src/main/kotlin/streams/utils/JSONUtils.kt:239]: maybe this then?

```kotlin
fun Map<String, Any?>.toMapValue(allowNulls: boolean): MapValue {
    return this
        .filterValues { v -> allowNulls || v != null }
        .entries
        .fold(MapValueBuilder()) { builder, entry -> 
            builder.add(entry.key, Values.of(entry.value, allowNulls)); builder}
        .build()
}
```

File: common/src/test/kotlin/streams/utils/JSONUtilsTest.kt (hunk 1 above)
Comment (fbiville) [reply common/src/test/kotlin/streams/utils/JSONUtilsTest.kt:130]: ah yes, sorry, got confused about the inputs and outputs...
I think splitting the test in 2 as suggested below and a couple of renames and/or explicit `//when`/`//then` comments would help here

**Review by fbiville — CHANGES_REQUESTED** [review]

File: common/src/test/kotlin/streams/utils/JSONUtilsTest.kt (hunk 1 above)
Comment (Emrehzl94) [reply common/src/test/kotlin/streams/utils/JSONUtilsTest.kt:130]: 👍

File: common/src/main/kotlin/streams/utils/JSONUtils.kt (hunk 3 above)
Comment (Emrehzl94) [reply common/src/main/kotlin/streams/utils/JSONUtils.kt:239]: yes, I think this works

**Review by fbiville — APPROVED** [review]

**Pull request merged**